src/Symfony/Component/Process/Process.php

```diff
@@ -429,12 +429,31 @@ public function wait(?callable $callback = null): int
         do {
             $this->checkTimeout();
             $running = $this->isRunning() && ('\\' === \DIRECTORY_SEPARATOR || $this->processPipes->areOpen());
-            $this->readPipes($running, '\\' !== \DIRECTORY_SEPARATOR || !$running);
+            if ($fiber = \Fiber::getCurrent()) {
+                $this->readPipes(false, '\\' !== \DIRECTORY_SEPARATOR || !$running);
+                $startedAt = microtime(true);
+                $fiber->suspend();
+                $sleepFor = (int) (1000 - (microtime(true) - $startedAt) * 1000000);
+                if (0 < $sleepFor) {
+                    usleep($sleepFor);
```

Comment:
maybe it could use stream_select on the pipes instead of usleep ? so it would directly read / write if there is something to do instead of sleeping ?

Reply:
we do use stream_select already so I guess we added this to address a situation where stream_select doesn't apply. git blame could help remember :)